moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1899,10 +1899,40 @@ double moveit::core::RobotState::computeCartesianPath(const JointModelGroup* gro
           computeCartesianPath(group, traj, link, target_pose, true, max_step, jump_threshold, validCallback, options));
 }
 
+double moveit::core::RobotState::computeCartesianPath(const JointModelGroup* group, std::vector<RobotStatePtr>& traj,
+                                                      const LinkModel* link, const Eigen::Vector3d& direction,
+                                                      bool global_reference_frame, double distance, double max_step,
+                                                      double jump_threshold,
+                                                      const GroupStateValidityCallbackFn& validCallback,
+                                                      const kinematics::KinematicsQueryOptions& options)
+{
+  jump_threshold_t jt;
+  jt.jump_threshold_factor = jump_threshold;
+  jt.prismatic_jump_threshold = 0.0;
+  jt.revolute_jump_threshold = 0.0;
+  return computeCartesianPath(group, traj, link, direction, global_reference_frame, distance, max_step, jt,
+                              validCallback, options);
+}
+
+double moveit::core::RobotState::computeCartesianPath(const JointModelGroup* group, std::vector<RobotStatePtr>& traj,
+                                                      const LinkModel* link, const Eigen::Vector3d& direction,
+                                                      bool global_reference_frame, double distance, double max_step,
+                                                      double revolute_jump_threshold, double prismatic_jump_threshold,
+                                                      const GroupStateValidityCallbackFn& validCallback,
+                                                      const kinematics::KinematicsQueryOptions& options)
+{
+  jump_threshold_t jt;
+  jt.jump_threshold_factor = 0.0;
+  jt.prismatic_jump_threshold = prismatic_jump_threshold;
+  jt.revolute_jump_threshold = revolute_jump_threshold;
+  return computeCartesianPath(group, traj, link, direction, global_reference_frame, distance, max_step, jt,
+                              validCallback, options);
+}
+
 double moveit::core::RobotState::computeCartesianPath(const JointModelGroup* group, std::vector<RobotStatePtr>& traj,
```

Comment:
this function can be private... do we want users accessing computeCartesianPath() with the struct? If we do, then maybe we don't need the ``double revolute_jump_threshold, double prismatic_jump_threshold,`` version

Reply:
Good question. I wrote it like this to allow the user to use the struct version to filter with both the jump_threshold factor method and the absolute jump thresholds. Although, on further reflection, I don't know if there is much of a use case for that.

Reply:
are you saying its possible to use both the old and new filtering methods at the same time? that was not my understanding of the code

Reply:
It was. I'm changing it to be one or the other.